lib/paperclip_processors/custom_file_preview.rb

```diff
@@ -11,25 +12,29 @@ def make
 
       begin
         if @file.content_type == 'application/pdf'
-          # We use special convert options for PDFs to improve quality and
-          # background, we append [0] to convert only the first page
-          convert(
-            ":source -resize '#{options[:geometry]}' -format #{options[:format]} -flatten -quality 70 :dest",
-            source: File.expand_path(@file.path) + '[0]',
-            dest: File.expand_path(dst.path)
+          content = Paperclip.run(
```

Comment:
If you provide base filename as last argument, pdftoppm will save image directly to that file and there will be no need to load everything to memory

Reply:
Was a long a day, thanks.